tests/TestBrowser.cpp

```diff
@@ -429,3 +430,57 @@ void TestBrowser::testValidURLs()
         QCOMPARE(Tools::checkUrlValid(i.key()), i.value());
     }
 }
+
+void TestBrowser::testBestMatchingCredentials()
+{
+    auto db = QSharedPointer<Database>::create();
+    auto* root = db->rootGroup();
+
+    // Test with simple URL entries
+    QStringList urls = {"https://github.com/loginpage", "https://github.com/justsomepage", "https://github.com/"};
+
+    auto entries = createEntries(urls, root);
+
+    browserSettings()->setBestMatchOnly(true);
+
+    auto result = m_browserService->searchEntries(db, "https://github.com/loginpage", "https://github.com/loginpage");
+    QCOMPARE(result.size(), 1);
+    QCOMPARE(result[0]->url(), QString("https://github.com/loginpage"));
```

Comment:
Oh, btw, I don't think you need to do these explicit constructor calls anymore. That was a thing that was necessary for Qt 5.2, but Qt 5.5 should be able to handle const char*.